### PR TITLE
Revert "Made grpc_core::Mutex compatible to absl::Mutex"

### PR DESCRIPTION
**Pull request by donnadionne**
This reverts commit 0b53341328d5d1a8047037622b622d2ddb50d5b1.

@donnadionne

### CODE REVIEW

**Review by markdroth — APPROVED** [review]
Esun should review as well.

**Comment by donnadionne** [comment]
Reverting 3 PRs in total:
Revert "Made grpc_core::Mutex compatible to absl::Mutex" …
Revert "Added thread annotation (#25486)" …
Revert "Fix a race in resolve_address and resolve_address_posix tests" …

File: test/core/iomgr/resolve_address_posix_test.cc [hunk 1]
```diff
@@ -96,24 +95,25 @@ static void actually_poll(void* argsp) {
   grpc_millis deadline = n_sec_deadline(10);
   while (true) {
     grpc_core::ExecCtx exec_ctx;
-    {
-      grpc_core::MutexLockForGprMu lock(args->mu);
```
Comment (apolcyn) [inline test/core/iomgr/resolve_address_posix_test.cc:100]: can we change this to just `grpc_core::MutexLock`. Same elsewhere for this file and `test/core/iomgr/resolve_address_test.cc`

**Review by apolcyn — CHANGES_REQUESTED** [review]
Please don't revert https://github.com/grpc/grpc/pull/25398 as a part of this.

Instead, I think we can just replace all instances of `grpc_core::MutexLockForGprMu` in `test/core/iomgr/resolve_address_test.cc` and `test/core/iomgr/resolve_address_posix_test.cc` with `grpc_core::MutexLock`

File: test/core/iomgr/resolve_address_posix_test.cc (hunk 1 above)
Comment (donnadionne) [reply test/core/iomgr/resolve_address_posix_test.cc:100]: OK, fixed

**Review by apolcyn — APPROVED** [review]
LGTM for changes `resolve_address_test.cc` and `resolve_address_posix_test.cc`

**Review by veblush — APPROVED** [review]

**Pull request merged**